### PR TITLE
Avoid running SQLAlchemy schema tests with Ingres dialect

### PR DESCRIPTION
**Pull request by hab6**
When running the SQLAlchemy [dialect compliance suite](https://github.com/sqlalchemy/sqlalchemy/tree/main/lib/sqlalchemy/testing/suite), quite a few tests would fail with some form of the following error:

    sqlalchemy.exc.ProgrammingError: (pyodbc.ProgrammingError)
    ('42503', "[42503] [Actian][Actian II ODBC Driver][INGRES]
    CREATE TABLE: You may not create an object owned by 'test_schema'. (328737) (SQLExecDirectW)")
    [SQL:
    CREATE TABLE test_schema.test_table (
        id INTEGER GENERATED BY DEFAULT AS IDENTITY NOT NULL,
        data VARCHAR(50),
        PRIMARY KEY (id) )

The reason for the error is that the affected tests attempt to create and drop tables owned under schemas different than the current user.

Ingres does not support this kind of operation except in rare instances where the current user has security administrator and operator privilege or who is the DBA of the database. The user must then use the `SET SESSION AUTHORIZATION ...` command to set the effective user for the current session.

Per discussion in [11366](https://github.com/sqlalchemy/sqlalchemy/discussions/11366) and additional research in internal ticket [II-14148](https://actian.atlassian.net/browse/II-14148), the setting `supports_schemas = False` will be added to the IngresDialect class to avoid execution of tests that use alternate schemas.

The file **README.testsuite.md** is being added to provide information on how to run the [dialect compliance suite](https://github.com/sqlalchemy/sqlalchemy/tree/main/lib/sqlalchemy/testing/suite) and also provide information about the issue of running tests that contain alternate schemas.

### Testing

    Ingres 11.2 (15807) on Windows 10 (client and server are same machine)
    Python 3.10.7
    mock 5.1.0
    pluggy 1.5.0
    psycopg2 2.9.9
    pyodbc 5.1.0
    pypyodbc 1.3.6
    pytest 8.1.1
    setuptools 69.5.1
    SQLAlchemy 2.0.29.dev0
    sqlalchemy-ingres 0.0.7.dev1
    tox 4.14.2

Result | Baseline | With<br>supports_schemas = False | Diff  
-- | -- | -- | --
Total Tests | 1531 | 1531 | -  
Passed | 336 | 473 | +137  
Failed | 63 | 173 | +110  
Errors | 798 | 30 | -768  
Skipped | 334 | 855 | +521  
Run Time | 17m 25s | 42s | -16m 43s

### CODE REVIEW

**Comment by hab6** [comment]
I am suspicious about the huge jump in skipped tests when setting `supports_schemas = False` in the IngresDialect class.

Result | Without<br>`supports_schemas = False` | With<br>`supports_schemas = False` | Diff
-- | -- | -- | --
Skipped | 334 | 855 | +521

I think it will be important to run the suite with and without the setting and review the 521 tests that are affected to see they are indeed skipped due to the alternate schema issue and not some side effect.

File: README.testsuite.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+### Known Issues
+
+#### Use of Alternate Schemas
+Several tests require alternate usernames or schemas to be present, which are used to test dotted-name access scenarios. On some databases such as Oracle these are usernames, and others such as PostgreSQL and MySQL they are schemas. The requirement applies to all backends except SQLite and Firebird. The names are:
+
+    test_schema
+    test_schema_2 (only used on PostgreSQL and mssql)
+
+Please refer to your vendor documentation for the proper syntax to create these namespaces - the database user must have permission to create and drop tables within these schemas. Its perfectly fine to run the test suite without these namespaces present, it only means that a handful of tests which expect them to be present will fail.
```
Comment (clach04) [inline README.testsuite.md:60]: quote these too, makes it more clear whats a quote (and where from) and our notes

File: README.testsuite.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+Added a new flag to .DefaultDialect called supports_schemas; third party dialects may set this flag to False to disable SQLAlchemy's schema-level tests when running the test suite for a third party dialect.
+
+_(The above text about alternate schemas is taken from the following SQLAlchemy GitHub pages)_
+
+[README.unittests.rst](https://github.com/sqlalchemy/sqlalchemy/blob/main/README.unittests.rst)  
+[changelog_14.rst](https://github.com/sqlalchemy/sqlalchemy/blob/main/doc/build/changelog/changelog_14.rst)  
+
+**Ingres Dialect Behavior**  
+The Ingres dialect disables the ability to run tests that use alternate schemas with the setting `supports_schemas = False` in class `IngresDialect`.
```
Comment (clach04) [inline README.testsuite.md:70]: This is a good statement, it needs an explanation adding (steal the text from your PR, that text is perfect).

File: lib/sqlalchemy_ingres/base.py [hunk 3]
```diff
@@ -387,6 +387,7 @@ class IngresDialect(default.DefaultDialect):
     supports_empty_insert = False
     supports_empty_insert = False
     supports_statement_cache = False  # NOTE this is not actually picked up by SA warning code, _generate_cache_attrs() checks dict of subclass, not the entire class
+    supports_schemas = False
```
Comment (clach04) [inline lib/sqlalchemy_ingres/base.py:390]: Recommend having a comment here, could simple say "see readme.testsuite.md for details"

**Review by clach04 — CHANGES_REQUESTED** [review]
Looks good.

Needs explanation text (that you wrote in the PR ) in the readme.

Minor tweak suggested for quoting.

**Comment by hab6** [comment]
@clach04 Thanks for your review. I added text from the PR to the new README and cleaned it up a bit, including your suggestions for making the quoted text distinct that is taken from the SQLAlchemy repository pages.

Please let me know of additional suggestions. Thanks.

**Review by clach04 — APPROVED** [review]

**Comment by hab6** [comment]
Focusing solely on the tests in dialect compliance suite class **ComponentReflectionTest**:

Without the flag `supports_schemas = False` in the **IngresDialect** class, all reflection tests fail with this error _(e.g. involving tables)_: **`[INGRES]CREATE TABLE: You may not create an object owned by 'test_schema'.`** Conversely, with the flag present in the **IngresDialect** class, the majority of the tests are skipped, yet many tests now fall into the `PASS` and `FAIL` categories.

**Results of running ComponentReflectionTest class**
IngresDialect class<br>contains flag<br>`supports_schemas = False` | PASSED | FAILED | ERROR | SKIPPED | TOTAL
--|--|--|--|--|--
No | 0 | 0 | 732 | 0 | 732
Yes | 131 | 104 | 0 | 497 | 732

**Pull request merged**